pkg/queue/readiness/probe.go

```diff
@@ -78,13 +77,23 @@ func (gv *gateValue) read() bool {
 	return gv.result
 }
 
-// NewProbe returns a pointer a new Probe
-func NewProbe(ctx context.Context, v1p *corev1.Probe) *Probe {
+// NewProbe returns a pointer to a new Probe.
+func NewProbe(v1p *corev1.Probe) *Probe {
 	return &Probe{
 		Probe:       v1p,
 		pollTimeout: PollTimeout,
 		out:         os.Stderr,
-		ctx:         ctx,
+	}
+}
+
+// NewProbeWithHTTP2AutoDetection returns a pointer to a new Probe that has HTTP2
+// auto-detection enabled.
+func NewProbeWithHTTP2AutoDetection(v1p *corev1.Probe) *Probe {
```

Comment:
any reason not to just add a bool param to the existing method? I think that way we can avoid an if/else at the call site too.

Reply:
I had to update a gazillion callsites in tests and it seemed "cooler" pushing the if/else all the way up. Not a hard opinion though.

Reply:
meh, if its painful I dont care much either - I just did a [quick search](https://github.com/knative/serving/search?q=NewProbe) and it seemed easy or I wouldnt have suggested but I guess github search is being misleading 🤷 . Im happy enough as is, maybe Ill fix up in a follow on at some point, no big deal.